spacecmd/src/spacecmd/utils.py

```diff
@@ -154,17 +149,29 @@ def load_cache(cachefile):
     else:
         logging.debug('%s does not exist', cachefile)
 
+    if isinstance(expire, str):
+        expire = datetime.strptime(expire, _CACHE_DATE_FORMAT)
+
     return data, expire
 
 
 def save_cache(cachefile, data, expire=None):
+    if expire:
+        if not isinstance(expire, (datetime, str)):
+            raise ValueError("expire must be datetime.datetime or str object")
+        try:
+            datetime.strptime(str(expire), _CACHE_DATE_FORMAT)
+        except ValueError as e:
+            raise ValueError(f"Provided expire parameter must conform to {_CACHE_DATE_FORMAT}") from e
+
     if expire:
         data['expire'] = expire
+    json_data = json.dumps(data, default=str, indent=4)
```

Comment:
[nitpick] Same question, we don't do anything with `json_data`, could be moved into the context manager and be `json.dump(data, f, ...)`